docs/architecture/0008-flask-site-security.md

```diff
@@ -0,0 +1,87 @@
+# 0003. Flask site security
+
+Date: 2021-09-03
+
+## Status
+
+* Proposed (2021-10-11)
+
+## Context
+
+As we go live with the Flask part of service-front, we need to
+ensure that this is secure enough, immediately for static pages
+and ultimately for forms as and when the first form is rolled out
+
+## Decision
+
+We will follow OWASP guidelines to secure the flask site.
+We will immediately do any obvious security protection.
+We will scan the python code
+
+### OWASP 2021 top 10 on flask site
+
+A01:2021 Broken Access Control :
+The initial release consists of static pages that are public. The first form will also be public. Ultimately, we aim to share session information with the php site. At this point tests will be required to ensure relevant content cannot be accessed without logging in.
+Except for public resources, deny by default.  This does happen to some extent in that the flask site needs nginx to actively be configured to proxy_pass pages to it. However we need to be careful with any wildcard instructions to nginx.
+
+A02:2021 Cryptographic failure, (formerly known as Sensitive Data Exposure).
+We avoid having any secrets in the repo.
```

Comment:
I would say this more forcefully, as this is MoJ policy:  `we do not keep any secrets in the repository and any in use will be protected using secrets management services, and accessed by code from there`. or along the lines of this. 
you may want to refer to this in general as well in this ADR: https://docs.opg.service.justice.gov.uk/documentation/guides/security_process.htmlhttps://docs.opg.service.justice.gov.uk/documentation/guides/security_process.html

Reply:
you might want to add: `we use security scans and precommit hooks to check for secrets, in order to reduce the chance of these being committed`

Reply:
All helpful :) Have added this.